src/ui/utils/helpers.ts

```diff
@@ -30,3 +31,23 @@ export function isValidTeamName(name: string) {
 
   return true;
 }
+
+export function extractIdAndSlug(params: string | string[] | undefined) {
```

Comment:
This is probably just a style thing, but to me the fact that next router has so many possible return types and the fact that we're looking at a variety of urls are problems that are separate and should be handled separately. I'm thinking something like:

```
export function toArray(params: string | string[] | undefined): string[] {
  if (Array.isArray(params)) {
    return params;
  } else if (params === undefined) {
    return [];
  } else {
    return Array(params);
  }
}

export function extractIdAndSlug(params: string | string[] | undefined) {
  let asArray = toArray(params);
  const id: string | undefined = toArray(asArray).find(validateUUID);
  const slug = asArray.length === 2 ? asArray[0] : undefined;

  return { id, slug };
}
```

Reply:
I considered something along this line but wanted to be more prescriptive about the URL format being either `/id` or `/slug/id` (and not `/slug/random/stuff/id`). There's probably a compromise though that converts to an Array as you've suggested but retains the prescriptive structure.

Reply:
yeah I am 100% certain there is a clever, elegant way to do it, but this is readable enough, and that's what matters most 👍